File: benchmarks/javascript/sum_of_digits.js
function sumDigits(n) {
    let sum = 0;
    while (n > 0) {
        sum += n % 10;
        n = Math.floor(n / 10);
    }
    return sum;
}

function sumAllDigits() {
    let totalSum = 0;
    for (let i = 1; i <= 100000000; i++) {
        totalSum += sumDigits(i);
    }
    console.log(`Sum of all digits from 1 to 100,000,000: ${totalSum}`);
}

sumAllDigits();

File: benchmarks/go/sum_of_digits.go
package main

import "fmt"

func sumDigits(n int64) int64 {
    var sum int64 = 0
    for n > 0 {
        sum += n % 10
        n /= 10
    }
    return sum
}

func main() {
    var totalSum int64 = 0
    for i := int64(1); i <= 100000000; i++ {
        totalSum += sumDigits(i)
    }
    fmt.Printf("Sum of all digits from 1 to 100,000,000: %d\n", totalSum)
}